### ISSUE TITLE
initialLayout behaves differently with and without js remote dedbug

### ISSUE THREAD

**Issue by zhukovka** [react-native-community/react-native-tab-view#330]
If state.index is set to a number other than 0, and an initial layout is set - TabViewAnimated works well with NO js remote debug, breaks with js remote debug on.
### Expected behaviour
the active tab is set to 1 and I can switch tabs both with and without remote js debugging
### Actual behaviour
the active tab is set to 1 and I can't switch tabs with remote js debug on - nothing happens when I'm trying to switch to another tab;

### Code sample, screenshots
```
class MissionComponent extends Component<MissionProps, IMissionState> {

    constructor (props) {
        super(props);

        this.state = {
            index : 1,
            routes : [
                {key : "map", title : "map"},
                {key : "streams", title : "streams"}
            ]
        }
    };
  
 _handleChangeTab = index => this.setState({index});
  render () {
        return (
            <View style={{flex : 1}}>
                <TabViewAnimated
                    renderScene={this._renderTab}
                    navigationState={this.state}
                    onRequestChangeTab={this._handleChangeTab}
                    changeRoute={this._handleChangeTab}
                    renderHeader={this._renderHeader}
                    initialLayout={{
                        height : 0,
                        width : Dimensions.get('window').width,
                    }}
                />
            </View>
        );
    }
}
```
![Imgur](http://i.imgur.com/tqkgc1g.gif)

**Comment by satya164** [comment]
Looks similar to this https://github.com/facebook/react-native/issues/13463

**Comment by zhukovka** [comment]
@satya164 doesn't seem to be related

**Comment by satya164** [comment]
You're sure? Are you able to swipe to change tabs?

**Comment by zhukovka** [comment]
@satya164 I am able when "Debug JS remotely" is off, it seems to be somehow related with "Debug JS remotely".

**Comment by satya164** [comment]
@zhukovka I understand that it only happens when you are connected to the debugger. But the behaviour you are describing looks very similar to the issue I linked to, despite the different steps to repro. Or you are saying the behaviour is different from that issue?

**Comment by itswaze** [comment]
I have a similar problem..

When Remote Debug is ON, my layout is correct: 
https://imgur.com/a/5YWyE

When Remote Debug is OFF, my layouts gets weird:
https://imgur.com/a/YWDEu

**Comment by zhukovka** [comment]
What I have found is in TabViewPagerScroll.js 
``` 
 _handleScroll = (e: ScrollEvent) => {
    this._isIdle =
      Math.abs(e.nativeEvent.contentOffset.x - this._nextOffset) < 0.1;
    this.props.position.setValue(
      e.nativeEvent.contentOffset.x / this.props.layout.width
    );
  };
```
in debug mode `e.nativeEvent.contentOffset.x` is 0 while in non-debugging mode `e.nativeEvent.contentOffset.x` is actual offset value
this is the reason I have my problem

**Comment by taiki-t** [comment]
This problem is maybe of react native itself: https://github.com/facebook/react-native/issues/15808
Other components that use ListView are also having similar problems.

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.
